### PR TITLE
Allow BabylonJS playground to load

### LINKED ISSUE #25478: Implement window "current event"
Lazily-written event listeners can be written by page authors as Javascript functions taking no arguments, trusting that the global "event" will be the right thing when they are called. https://dom.spec.whatwg.org/#interface-window-extensions is what makes this possible.
This might be easier to implement after I finish doing some cleanup for #22783; I'll be sure to leave comments for the relevant points.

### LINKED ISSUE #26428: Support the babylon playground
We should identify what's missing in our implementation to support the babylon playground, which is a common way for developers to experiment with new content.

### PR DESCRIPTION
**Pull request by jdm**
These changes fix the JS errors that currently appear when loading the page. You can't use the editor, but the webgl canvas works just fine.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #26428 and fix #25478
- [x] There are tests for these changes

### CODE REVIEW

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /dom/events/event-global.html:
  └ PASS [expected FAIL] window.event is set to the current event during dispatch
  ▶ Unexpected subtest result in /dom/events/event-global.html:
  └ PASS [expected FAIL] window.event is set to the current event, which is the event passed to dispatch
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=(Document|Window):
  └ PASS [expected FAIL] Document interface: operation queryCommandSupported(DOMString)
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=(Document|Window):
  └ PASS [expected FAIL] Document interface: iframe.contentDocument must inherit property "queryCommandSupported(DOMString)" with the proper type
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=(Document|Window):
  └ PASS [expected FAIL] Document interface: calling queryCommandSupported(DOMString) on iframe.contentDocument with too few arguments must throw TypeError
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=(Document|Window):
  └ PASS [expected FAIL] Document interface: new Document() must inherit property "queryCommandSupported(DOMString)" with the proper type
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=(Document|Window):
  └ PASS [expected FAIL] Document interface: calling queryCommandSupported(DOMString) on new Document() with too few arguments must throw TypeError
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=(Document|Window):
  └ PASS [expected FAIL] Document interface: documentWithHandlers must inherit property "queryCommandSupported(DOMString)" with the proper type
  ▶ Unexpected subtest result in /html/dom/idlharness.https.html?include=(Document|Window):
  └ PASS [expected FAIL] Document interface: calling queryCommandSupported(DOMString) on documentWithHandlers with too few arguments must throw TypeError
  ▶ Unexpected subtest result in /dom/idlharness.window.html?exclude=Node:
  └ PASS [expected FAIL] Window interface: attribute event
```

File: components/script/dom/event.rs [hunk 1]
```diff
@@ -690,12 +690,15 @@ fn inner_invoke(
...
-        // FIXME(#25478): we need to get the global that the event
-        // listener is going to be called on, then if it's a Window
-        // set its .event to the event, remembering the previous
-        // value of its .event. This allows events to just use
-        // the word "event" instead of taking the event as an argument.
+        // Step 2.6
+        let global = listener.associated_global();
+
+        // Step 2.7-2.8
+        let current_event = if let Some(window) = global.downcast::<Window>() {
```
Comment (Manishearth) [inline components/script/dom/event.rs:697]: i would perhaps call this `last_event` or something?

**Review by Manishearth — APPROVED** [review]
r=me either way

File: components/script/dom/event.rs (hunk 1 above)
Comment (jdm) [reply components/script/dom/event.rs:697]: I'm going with the name from the spec.

**Pull request merged**